subsys/jwt/CMakeLists.txt

```diff
@@ -2,4 +2,9 @@
 
 zephyr_library()
 zephyr_library_sources(jwt.c)
+
+zephyr_library_sources_ifdef(CONFIG_JWT_SIGN_ECDSA_LEGACY jwt_ecdsa.c)
+zephyr_library_sources_ifdef(CONFIG_JWT_SIGN_RSA_LEGACY jwt_rsa.c)
```

Comment:
IMO would make more sense to name those files something `legacy` if they're not holding the ECDSA/RSA implementation in all cases (in the PSA case it's in `jwt_psa.c`.